foundry/applications/formApplications/baseEntitySheets/rollTableConfig.d.ts

```diff
@@ -1,25 +1,151 @@
- *
- * @param table - The rollable table entity being configured
- * @param options - Additional application rendering options
  */
-declare class RollTableConfig extends BaseEntitySheet {
+declare class RollTableConfig extends BaseEntitySheet<RollTableConfig.Data, RollTable> {
+  static get defaultOptions(): RollTableConfig.Options;
+
   /**
-   * Return a dynamic application title for the configuration sheet
+   * @override
    */
   get title(): string;
 
   /**
-   * Activate event listeners for the RollTable Configuration Sheet
+   * @param options - (unused)
+   * @override
+   */
+  getData(options?: Application.RenderOptions): RollTableConfig.Data;
+
+  /**
+   * @override
    */
   activateListeners(html: JQuery): void;
 
   /**
-   * @param options - (unused)
+   * Handle creating a TableResult in the RollTable entity
+   * @param event      - The originating mouse event
+   * @param resultData - An optional object of result data to use
+   */
+  protected _onCreateResult(event: MouseEvent, resultData?: object): Promise<any>; // TODO: type when TableResult is typed
+
+  /**
+   * Submit the entire form when a table result type is changed, in case there are other active changes
+   */
+  protected _onChangeResultType(event: Event): ReturnType<RollTableConfig['_onSubmit']>;
+
+  /**
+   * Handle deleting a TableResult from the RollTable entity
+   */
+  protected _onDeleteResult(event: Event): Promise<any>; // TODO: type when TableResult is typed
+
+  /**
    * @override
    */
-  getData(options?: any): BaseEntitySheet.Data<Entity>;
+  protected _onDrop(event: DragEvent): ReturnType<RollTableConfig['_onCreateResult']>;
+
+  /**
+   * Handle changing the actor profile image by opening a FilePicker
+   */
+  protected _onEditImage(event: Event): ReturnType<FilePicker['browse']>;
+
+  /**
+   * Handle a button click to re-normalize dice result ranges across all RollTable results
+   */
+  protected _onNormalizeResults(event: Event): Promise<any>; // TODO: type when TableResult is typed
+
+  /**
+   * Handle toggling the drawn status of the result in the table
+   */
+  protected _onLockResult(event: Event): Promise<any>; // TODO: type when TableResult is typed
+
+  /**
+   * Reset the Table to it's original composition with all options unlocked
+   */
+  protected _onResetTable(event: Event): Promise<any>; // TODO: type when TableResult is typed
+
+  /**
+   * Handle drawing a result from the RollTable
+   */
+  protected _onRollTable(event: Event): Promise<void>;
+
+  /**
+   * Configure the update object workflow for the Roll Table configuration sheet
+   * Additional logic is needed here to reconstruct the results array from the editable fields on the sheet
+   * @param event    - The form submission event
+   * @param formData - The validated FormData translated into an Object for submission
+   */
+  protected _updateObject(event: Event, formData: RollTableConfig.Data): Promise<any>; // TODO: type when TableResult is typed
```

Comment:
`RollTableConfig.Data` is not the right type here. I tried it and the `formData` I received was
```ts
{
    description: "test123",
    displayRoll: true,
    formula: "",
    img: "icons/svg/d20-grey.svg",
    name: "Test",
    replacement: true,
    results.0._id: "vFglpGu6G6IsA0jj",
    results.0.drawn: false,
    results.0.img: "icons/svg/d20-black.svg",
    results.0.rangeH: 1,
    results.0.rangeL: 1,
    results.0.resultId: "",
    results.0.text: "fghdf",
    results.0.type: 0,
    results.0.weight: 1,
    results.1._id: "SjkYR07Ctogz3z8W",
    results.1.drawn: false,
    results.1.img: "icons/svg/d20-black.svg",
    results.1.rangeH: 2,
    results.1.rangeL: 2,
    results.1.resultId: "",
    results.1.text: "dfg",
    results.1.type: 0,
    results.1.weight: 1,
    results.2._id: "r58Xh0s00UQH2jI4",
    results.2.drawn: false,
    results.2.img: "icons/svg/d20-black.svg",
    results.2.rangeH: 3,
    results.2.rangeL: 3,
    results.2.resultId: "",
    results.2.text: "dfg",
    results.2.type: 0,
    results.2.weight: 1,
}
```
Again, seems to correspond to the editable stuff in the form. Looks like `flattenObject` was called on the data somewhere inbetween though, at least it is a flattened object.